src/Sparrow/Platform/Posix/Syscall.cs

```diff
@@ -260,7 +252,18 @@ public static long lseek64(int fd, long offset, WhenceFlags whence)
         public static extern int statvfs(string path, ref Statvfs buf);
 
         [DllImport(LIBC_6, SetLastError = true)]
-        public static extern int mprotect(IntPtr start, ulong size, ProtFlag protFlag);
+        public static extern int mprotect_64bit(IntPtr start, ulong size, ProtFlag protFlag);
+        
+        [DllImport(LIBC_6, SetLastError = true)]
+        public static extern int mprotect_32bit(IntPtr start, IntPtr size, ProtFlag protFlag);
```

Comment:
No, this is _always_ `IntPtr`, no need for both of them. And you didn't set the name, so this would always fail